packages/flutter/lib/src/widgets/overscroll_indicator.dart

```diff
@@ -834,138 +826,238 @@ class _StretchingOverscrollIndicatorState extends State<StretchingOverscrollIndi
   }
 }
 
-enum _StretchState { idle, absorb, pull, recede }
+/// A physics simulation modeling the stretch behavior of an edge effect.
+///
+/// This class is a direct port of Android's EdgeEffect.java:
+/// https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/core/java/android/widget/EdgeEffect.java
+///
+/// It reproduces the damped harmonic oscillator behavior of Android's
+/// overscroll effect in Flutter, using the same physical constants:
+/// - natural frequency
+/// - damping ratio
+///
+/// The formulas and coefficients are derived directly from the Android source.
+class _StretchSimulation extends Simulation {
```

Comment:
Can you replace this with `SpringSimulation` as in the `spring2` in [this comment](https://github.com/flutter/flutter/pull/173849#issuecomment-3310730196)?